### PR TITLE
Consistently use feature name floating versions

### LINKED ISSUE #1861: Floating versions should not be called wildcard versions
Floating versions should not be called wildcard versions. 

That added with the usage of `*` makes people think that it's related to regex, but floating versions have their own semantics.

### PR DESCRIPTION
**Pull request by zivkan**
fixes #1861 

Note: since I changed a heading, it will break any link that goes to that anchor on the page. I updated all references within our own docs, but any links on other docs, blogs, stack overflow, and so on, will all be broken if this is merged.

### CODE REVIEW

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[4df0951](https://github.com/NuGet/docs.microsoft.com-nuget/commits/4df0951b586d0c74ffc7fb5f862cbac73d082750)_: 

:clock10: **Pending**: waiting for processors ([5](https://ops.microsoft.com/#/statistic) builds ahead of you) 

 :warning: **Docs Build is busy**, currently there are 5 builds ahead of this one, for more information you can view the Build queue graph on the [Docs Portal](https://ops.microsoft.com/#/statistic).

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[4df0951](https://github.com/NuGet/docs.microsoft.com-nuget/commits/4df0951b586d0c74ffc7fb5f862cbac73d082750)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/api/registration-base-url-resource.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/zivkan-use-floating-version/docs/api/registration-base-url-resource.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/api/registration-base-url-resource?branch=pr-en-us-1919) |
[docs/concepts/Dependency-Resolution.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/zivkan-use-floating-version/docs/concepts/Dependency-Resolution.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/concepts/dependency-resolution?branch=pr-en-us-1919) |
[docs/concepts/Package-Versioning.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/zivkan-use-floating-version/docs/concepts/Package-Versioning.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/concepts/package-versioning?branch=pr-en-us-1919) |
[docs/consume-packages/Package-References-in-Project-Files.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/zivkan-use-floating-version/docs/consume-packages/Package-References-in-Project-Files.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files?branch=pr-en-us-1919) |
[docs/consume-packages/Reinstalling-and-Updating-Packages.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/zivkan-use-floating-version/docs/consume-packages/Reinstalling-and-Updating-Packages.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/consume-packages/reinstalling-and-updating-packages?branch=pr-en-us-1919) |
[docs/reference/nuspec.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/zivkan-use-floating-version/docs/reference/nuspec.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/reference/nuspec?branch=pr-en-us-1919) |
[docs/reference/packages-config.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/zivkan-use-floating-version/docs/reference/packages-config.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/reference/packages-config?branch=pr-en-us-1919) |
[docs/release-notes/NuGet-1.4.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/zivkan-use-floating-version/docs/release-notes/NuGet-1.4.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/release-notes/nuget-1.4?branch=pr-en-us-1919) |
[docs/what-is-nuget.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/zivkan-use-floating-version/docs/what-is-nuget.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/what-is-nuget?branch=pr-en-us-1919) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C28%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C202002281849383374-1919%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=yIbVS%2Fgy6Bis0a85%2B01pAhOklmxocT9UMKmhlJyg0Zo%3D&st=2020-02-28T18%3A45%3A18Z&se=2020-03-30T18%3A50%3A18Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/api/registration-base-url-resource.md [hunk 1]
```diff
@@ -248,7 +248,7 @@ The alternate package object has the following properties:
 Name         | Type   | Required | Notes
 ------------ | ------ | -------- | -----
 id           | string | yes      | The ID of the alternate package
-range        | object | no       | The allowed [version range](../concepts/package-versioning.md#version-ranges-and-wildcards), or `*` if any version is allowed
+range        | object | no       | The allowed [version range](../concepts/package-versioning.md#version-ranges), or `*` if any version is allowed
```
Comment (donnie-msft) [inline docs/api/registration-base-url-resource.md]: ditto about "floating"

File: docs/concepts/Dependency-Resolution.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ When multiple packages have the same dependency, then the same package ID can ap
 
 ## Dependency resolution with PackageReference
 
-When installing packages into projects using the PackageReference format, NuGet adds references to a flat package graph in the appropriate file and resolves conflicts ahead of time. This process is referred to as *transitive restore*. Reinstalling or restoring packages is then a process of downloading the packages listed in the graph, resulting in faster and more predictable builds. You can also take advantage of wildcard (floating) versions, such as 2.8.\*, avoiding expensive and error prone calls to `nuget update` on the client machines and build servers.
+When installing packages into projects using the PackageReference format, NuGet adds references to a flat package graph in the appropriate file and resolves conflicts ahead of time. This process is referred to as *transitive restore*. Reinstalling or restoring packages is then a process of downloading the packages listed in the graph, resulting in faster and more predictable builds. You can also take advantage of floating versions, such as 2.8.\*,  to avoid modifying the project to use the latest version of a package.
```
Comment (donnie-msft) [inline docs/concepts/Dependency-Resolution.md]: Should we formalize the noun "Floating Versions", or bold/italic/etc to emphasize it?

File: docs/concepts/Dependency-Resolution.md [hunk 3]
```diff
@@ -49,16 +49,16 @@ When an application specifies an exact version number, such as 1.2, that is not
 
 <a name="floating-versions"></a>
 
-#### Floating (wildcard) versions
+#### Floating versions
 
-A floating or wildcard dependency version is specified with the \* wildcard, as with 6.0.\*. This version specification says "use the latest 6.0.x version"; 4.\* means "use the latest 4.x version." Using a wildcard allows a dependency package to continue evolving without requiring a change to the consuming application (or package).
+A floating dependency version is specified with the \* character. For example, `6.0.*`. This version specification says "use the latest 6.0.x version"; `4.*` means "use the latest 4.x version." Using a floating version reduces changes to the project file, while keeping up to date with the latest version of a dependency.
```
Comment (donnie-msft) [inline docs/concepts/Dependency-Resolution.md]: If emphasize or capitalization is changed by my other feedback, also change it here for "floating version".

File: docs/concepts/Dependency-Resolution.md [hunk 4]
```diff
@@ -49,16 +49,16 @@ When an application specifies an exact version number, such as 1.2, that is not
...
 
-When using a wildcard, NuGet resolves the highest version of a package that matches the version pattern, for example 6.0.\* gets the highest version of a package that starts with 6.0:
+When using a floating version, NuGet resolves the highest version of a package that matches the version pattern, for example `6.0.*` gets the highest version of a package that starts with 6.0:
```
Comment (donnie-msft) [inline docs/concepts/Dependency-Resolution.md]: ditto about "floating version" formatting

File: docs/api/registration-base-url-resource.md [hunk 5]
```diff
@@ -215,7 +215,7 @@ Each package dependency has the following properties:
 Name         | Type   | Required | Notes
 ------------ | ------ | -------- | -----
 id           | string | yes      | The ID of the package dependency
-range        | object | no       | The allowed [version range](../concepts/package-versioning.md#version-ranges-and-wildcards) of the dependency
+range        | object | no       | The allowed [version range](../concepts/package-versioning.md#version-ranges) of the dependency
```
[the remark this answers is not in the document]
Comment (zivkan) [reply docs/api/registration-base-url-resource.md]: The word floating is not used anywhere in `docs/api/registration-base-url-resource.md`. I don't understand your comment.

File: docs/concepts/Dependency-Resolution.md (hunk 2 above)
Comment (zivkan) [reply docs/concepts/Dependency-Resolution.md]: in my opinion, no.

File: docs/api/registration-base-url-resource.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (donnie-msft) [reply docs/api/registration-base-url-resource.md]: You're calling it a `version range` in this doc, but in Dependency-Resolution, you call it `floating version`.

[the remark this answers is not in the document]
Comment (zivkan) [reply docs/api/registration-base-url-resource.md]: floating versions are a sub-feature of version ranges. It's not mandatory to use floating versions with version ranges. In fact, considering floating ranges are not allowed in `nuspec` dependencies, this page's docs, on the v3 protocol, there should never be floating versions reported.

**Pull request merged**